### PR TITLE
Move ssh, bash, logtail, status, and run out of the deploy command hierarchy

### PR DESCRIPTION
**Pull request by apurvis**
Also:

- remove `:git_repo` config - unless i missed something it is not used anywhere in the code.
- rename `config.file` to `config.config_file` - i found this confusing
- display `Infinity` when polling forever instead of nothing
- default to `'ecs'` as a type until we have an actual non ECS option

not sure what version number to use here - i guess technically we should do 3.0.

### CODE REVIEW

**Comment by apurvis** [comment]
any thoughts?  cc @camertron @slpsys

**Comment by matl33t** [comment]
lgtm

**Comment by apurvis** [comment]
what version number should it be?

**Comment by apurvis** [comment]
(i think strictly speaking it probably has to be 3.0?  i'm not averse to major version bumps; they are informative)

**Comment by matl33t** [comment]
3 is fine, but are there any other features that break compatibility that we can fit in to this bump?

**Comment by apurvis** [comment]
i reviewed the [issues list](https://github.com/lumoslabs/broadside/issues) but i'm not super familiar with the feature requests to know if they are breaking.  any you notice?

File: lib/broadside/target.rb [hunk 1]
```diff
@@ -16,27 +16,16 @@ class Target
       :predeploy_commands,
       :scale,
       :service_config,
-      :tag,
```
Comment (apurvis) [inline lib/broadside/target.rb]: @matl33t actually this is an important question - i think in 2.0 i broke the ability to configure the `tag` per target - @camertron noticed that - and here i am fully removing that feature.  if we want that feature back, now would be the time.

**Comment by apurvis** [comment]
i'll just merge this; we can handle the version bump later

File: lib/broadside/target.rb (hunk 1 above)
Comment (matl33t) [inline lib/broadside/target.rb]: i think this is a feature we would like to keep

Comment (apurvis) [inline lib/broadside/target.rb]: whoops; i should have waited for your answer before merging, but i'll put it back in a separate PR

**Pull request merged**